core/server/src/main/java/alluxio/worker/netty/BlockDataServerHandler.java

```diff
@@ -80,14 +79,10 @@ void handleBlockReadRequest(final ChannelHandlerContext ctx, final RPCBlockReadR
     final long lockId = req.getLockId();
     final long sessionId = req.getSessionId();
 
-    BlockReader reader;
+    BlockReader reader = null;
     DataBuffer buffer;
     try {
       reader = mWorker.readBlockRemote(sessionId, blockId, lockId);
```

Comment:
I think the new logic is correct. We probably also want to do `req.validate` before anything else.